ya_glm/opt/constraint/convex.py

```diff
@@ -25,6 +26,22 @@ def _prox(self, x, step=1):
     def is_proximable(self):
         return True
 
+class LinearEquality(Constraint):
+    # credited to PyUNLocBoX
+    # https://github.com/epfl-lts2/pyunlocbox/
+    def __init__(self, A, b):
+        self.A = A
+        self.b = b
+        self.pinvA = np.linalg.pinv(A)
```

Comment:
Perhaps we should avoid computing this in __init__ (sometimes we initialize this object just to check if it is proximable). Maybe we should cache it the first time we call prox? I've also though about having a `.setup()` method that precomputes required data for these functions.

Reply:
we'll do it as a `@property`

Reply:
Fixed